lib/UserDiscoveryModules/IUserDiscoveryModule.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * @copyright Copyright (c) 2018 Bjoern Schiessle <bjoern@schiessle.org>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\GlobalSiteSelector\UserDiscoveryModules;
+
+interface IUserDiscoveryModule {
+
+	/**
+	 * get the initial user location
+	 *
+	 * @param array $data arbitrary data, whatever the module needs
```

Comment:
We should a bit more specific here. Something like "in case of SAML we hand in the raw data which contains usually ..." or "... which contains usually all SAML attributes that start with xyz_..."